prettyprinter/test/Testsuite/Main.hs

```diff
@@ -296,3 +298,9 @@ removeTrailingWhitespacePreserveIndentation
   = let sdoc :: SimpleDocStream ()
         sdoc = SLine 2 (SChar 'x' SEmpty)
     in assertEqual "" sdoc (removeTrailingWhitespace sdoc)
+
+regressionUnboundedGroupedLine :: Assertion
+regressionUnboundedGroupedLine
+  = let sdoc :: SimpleDocStream ()
+        sdoc = layoutPretty (LayoutOptions Unbounded) (group hardline)
+    in assertEqual "" (SLine 0 SEmpty) sdoc
```

Comment:
As noted in https://github.com/quchen/prettyprinter/issues/91#issuecomment-576610801, I should extract a more "robust" test case from `dhall`'s testsuite.

Reply:
Extracted from `escapeSingleQuotedOpenInterpolation`:

```
> layoutSmart (LayoutOptions Unbounded) $ group $ align $ "x" <> hardline <> "y"
SChar 'x' SFail
```